src/Core/View.js

```diff
@@ -44,7 +44,17 @@ function View(crs, viewerDiv, options = {}) {
 
     this.referenceCrs = crs;
 
-    this.mainLoop = options.mainLoop || new MainLoop(new Scheduler(), new c3DEngine(options.renderer || viewerDiv));
+    let engine;
+    // options.renderer can be 2 separate things:
+    //   - an actual renderer (in this case we don't use viewerDiv)
+    //   - options for the renderer to be created
```

Comment:
This comment should be in the doc above. Suggestion:

```js
/**
 * ...
 * @param {?(WebGLRenderer|object)} options.renderer - {@link WebGLRenderer} instance to use, otherwise 
 * a default one will be constructed. In this case, if options.renderer is an object, it will be used to 
 * configure the renderer (see {@link c3DEngine}.  If not present, a new <canvas> will be created and 
 * added to viewerDiv (mutually exclusive with mainLoop)
```